src/rgw/driver/sfs/sqlite/sqlite_lifecycle.cc

```diff
@@ -0,0 +1,97 @@
+// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t
+// vim: ts=8 sw=2 smarttab ft=cpp
+/*
+ * Ceph - scalable distributed file system
+ * SFS SAL implementation
+ *
+ * Copyright (C) 2023 SUSE LLC
+ *
+ * This is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License version 2.1, as published by the Free Software
+ * Foundation. See file COPYING.
+ */
+#include "sqlite_lifecycle.h"
+
+using namespace sqlite_orm;
+namespace rgw::sal::sfs::sqlite  {
+
+SQLiteLifecycle::SQLiteLifecycle(DBConnRef _conn) : conn(_conn) { }
+
+std::optional<DBOPLCHead> SQLiteLifecycle::get_head(const std::string & oid) const {
```

Comment:
This doesn't seem to need to be an `std::optional`. There isn't a branch that returns anything but an actual pointer?

Reply:
yup, it's no longer an optional. Because it is always returning a value.